File: Greedy/maximum69Number.js
const maximum69Number = (num) => {
  const digits = String(num).split("");
  for (let i = 0; i < digits.length; i++) {
    if (digits[i] === "6") {
      digits[i] = "9";
      break;
    }
  }

  return Number(digits.join(""));
};

// Time Complexity: O(n) where n is the number of digits in the input number
// Space Complexity: O(n) where n is the number of digits in the input number

File: Greedy/maximum_69_number.py
class Maximum69Number:
    def maximum69Number(self, num: int) -> int:
        digits = list(str(num))
        for i, digit in enumerate(digits):
            if digit == "6":
                digits[i] = "9"
                break
        return int("".join(digits))

    # Time complexity: O(d) where d is the number of digits in the input number.
    # In the worst case, we may need to check each digit.
    # Space complexity: O(d) since we are creating a list of digits from the input number.
